### ISSUE TITLE
Migrating to Bioghost (or other?)

### ISSUE THREAD

**Issue by gvegayon** [Yinqi93/LA-BEST-TA#2]
This is the discussion thread about where to host the tutorial @Yinqi93 built using Shiny. @pmarjora @jimb3 I'm adding you both to this thread b/c I think we could be able to host this on Bioghost. Right now, we do have a shiny server, but Bioghost is behind a firewall only for USC ppl, so students that participate, say, in the LA BEST program wouldn't be able to see it.

@zawtiger was very quick in creating a nice URL for the app, which is currently hosted on RStudio's shiny website, here https://iteams.usc.edu/r-shiny-tutorials/.

@jimb3 @pmarjora what are your thoughts about this? @zawtiger, we could also think about building a machine at prevmed just to host shiny apps...

**Comment by pmarjora** [comment]
I have no problem with us hosting this on Bioghost if that is a workable solution.

**Comment by zawtiger** [comment]
or host it on AWS (the future).
What are the requirements? (size, memory, OS, etc)

**Comment by gvegayon** [comment]
That's a good point! I think it is not that much right now (at most 10
people using it simultaneously at its peak?). The main issue will be when
more people start using it. Using AWS would be great, but I'm not sure how
do Shiny servers integrate with that (we may need to pay an extra fee?).

George G. Vega Yon
+1 (626) 381 8171
https://ggvy.cl

**Comment by Yinqi93** [comment]
Thanks for helping to host the R tutorial on our server! Currently the tutorial folder is 11 MB, although we might include some other materials in the future but it won't take too much memory. (I guess less 20 MB)

**Comment by gvegayon** [comment]
Yes! I think the main issue about memory is who is going to be using the
server. More than the disk space is how much memory will be used by the
server's sessions. So it could be that a single session uses, say, 200MB,
so that multiplied by 10, you have ~2G. Because of that reason, this should
be either hosted in a large server with other things going on, like
Bioghost, or in a smaller server by itself.

Best,

George G. Vega Yon
+1 (626) 381 8171
https://ggvy.cl

**Comment by gvegayon** [comment]
So we now have uscbiostats.shinyapps.io the credentials were shared via email. The actual app still needs to be uploaded. I've also suggested @Yinqi93 to build an R package of this, which would then avoid using a server for this (see #3)

**Comment by Yinqi93** [comment]
> So we now have uscbiostats.shinyapps.io the credentials were shared via email. The actual app still needs to be uploaded. I've also suggested @Yinqi93 to build an R package of this, which would then avoid using a server for this (see #3)

Thanks @gvegayon, I've successfully deployed the tutorial on our new uscbiostats shiny server! Now the website address is under uscbiostats account instead of my personal account. I just have one question: the link on LA BEST webpage still directs to the previous address (start with Yinqi93, see the "access to the tutorial" button on https://preventivemedicine.usc.edu/education/summer-programs/lasbest/). Do we need to setup a new url (starts with uscbiostats) for the tutorial? Thanks.

cc @ksiegmund

**Comment by gvegayon** [comment]
That's great, @Yinqi93! I think we need the help of @zawtiger for this. Andrew, could you update the URLs to link to this page instead?:

https://uscbiostats.shinyapps.io/Introduction_to_R/

These would be the link in 

1. https://preventivemedicine.usc.edu/education/summer-programs/lasbest/
2. https://iteams.usc.edu/r-shiny-tutorials/

I would actually update the button on (1) to point to (2), so that if we need to change the address again, we simply change (2). Does that make sense?

**Comment by gvegayon** [comment]
@ksiegmund I was checking the features that come with the free account, and it turns out that you only have 25 hours a month of usage, i.e., if 25 students connect to the app during one hour, shinyapps will stop working until next month. The next tier is 100USD/year and provides 100 hours. You should consider looking into that as my gut feeling tells me that you will run out of compute time pretty soon. (see [here](https://www.shinyapps.io/#pricing))
